src/Extension.php

```diff
@@ -43,7 +46,7 @@ public function load(ContainerBuilder $container, array $config): void
             [
                 new Reference(DefinitionExtension::FINDER_ID),
                 new Reference(DefinitionExtension::REPOSITORY_ID),
-                new Reference('unused_step_definitions_printer'),
+                new Reference($config['printer']),
```

Comment:
I believe adding a type cast should fix the error from PHPStan:
 
```suggestion
                new Reference((string) $config['printer']),
```